From: Alexander Larsson <alexl@redhat.com>
Date: Tue, 17 Jan 2017 16:36:56 +0100
Subject: Use seccomp to filter out TIOCSTI ioctl

This would otherwise let the sandbox add input to the controlling tty.

Origin: upstream, 0.8.1, commit:902fb713990a8f968ea4350c7c2a27ff46f1a6c4
---
 common/flatpak-run.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/common/flatpak-run.c b/common/flatpak-run.c
index 027c4a5..eff5a7b 100644
--- a/common/flatpak-run.c
+++ b/common/flatpak-run.c
@@ -26,6 +26,7 @@
 #include <unistd.h>
 #include <sys/utsname.h>
 #include <sys/socket.h>
+#include <sys/ioctl.h>
 #include <grp.h>
 
 #ifdef ENABLE_SECCOMP
@@ -3519,6 +3520,9 @@ setup_seccomp (GPtrArray  *argv_array,
     {SCMP_SYS (mount)},
     {SCMP_SYS (pivot_root)},
     {SCMP_SYS (clone), &SCMP_A0 (SCMP_CMP_MASKED_EQ, CLONE_NEWUSER, CLONE_NEWUSER)},
+
+    /* Don't allow faking input to the controlling tty (CVE-2017-5226) */
+    {SCMP_SYS (ioctl), &SCMP_A1(SCMP_CMP_EQ, (int)TIOCSTI)},
   };
 
   struct
